src/dmd/utils.d

```diff
@@ -210,6 +140,64 @@ void escapePath(OutBuffer* buf, const(char)* fname)
     }
 }
 
+/**
+ * Takes a path, and make it compatible with Makefile format.
+ *
+ * That is, turns
+ *  - '\' into '/'
+ *  - '$' into "$$"
+ *  - ' ' into "\ "
+ *  - '%' into "\%"
+ *  - '#' into "\#"
+ *  - ':' into "\:"
+ *
+ * Params:
+ *   buf = Buffer to write the escaped path to
+ *   fname = Path to escape
+ */
+void escapeMakePath(OutBuffer *buf, const(char)* fname)
+{
+    while(*fname)
+    {
+        switch (*fname)
+        {
+            case '\\':
+                buf.writeByte('/');
+                break;
+            case '$':
+                buf.writestring("$$");
+                break;
+            case ' ':
+                buf.writestring("\\ ");
+                break;
+            case '%':
+                buf.writestring("\\%");
+                break;
+            case '#':
+                buf.writestring("\\#");
+                break;
+            case ':':
+                buf.writestring("\\:");
+                break;
+            default:
+                buf.writeByte(*fname);
+                break;
+        }
+        fname++;
```

Comment:
```suggestion
    uint slashes;
    while(*fname)
    {
        switch (*fname)
        {
            case '\\':
                slashes++;
                break;
            case '$':
                buf.writeByte('$');
                goto default;
            case ' ':
            case '\t':
                while (slashes--)
                    buf.writeByte('\\');
                goto case;
            case '#':
            case ':':
                buf.writeByte('\\');
                goto default;
            default:
                slashes = 0;
                break;
        }
        buf.writeByte(*fname);
        fname++;
```

Reply:
Thank you for active support. It looks saner now.
I'm still a little bit puzzled by the escaping of `:` on windows paths (e.g. `C:\My Project` becomes `C\:\My\ Project`)

Reply:
> I'm still a little bit puzzled by the escaping of `:` on windows paths

I tried it with GNU make 4.3, "escaping" `:` seems to work, but not with 3.81. I doubt it works in any other kind of make, so I recommend not escaping it on Windows (it's a forbidden character in filenames anyway).
Please also add some Windows specific unittest.

Reply:
Yes, this seems the right thing to do.